agent/lib_guzzle6.c

```diff
@@ -458,33 +408,142 @@ void nr_guzzle6_enable(TSRMLS_D) {
       "      return $promise;"
       "    };"
       "  }"
-      "}",
-      NULL, "newrelic/Guzzle6" TSRMLS_CC);
+      "}", guzzle_version, guzzle_version, guzzle_version);
+
+  char *guzzle_ver = nr_formatf("newrelic/Guzzle%d", guzzle_version);
+  _retval = zend_eval_string(eval, NULL, guzzle_ver TSRMLS_CC);
 
-  if (SUCCESS == retval) {
+  if (SUCCESS == _retval && guzzle_version == 6) {
     nr_php_wrap_user_function(NR_PSTR("GuzzleHttp\\Client::__construct"),
-                              nr_guzzle_client_construct TSRMLS_CC);
-  } else {
+                              nr_guzzle6_client_construct TSRMLS_CC);
+  }else if (SUCCESS == _retval && guzzle_version == 7){
+    nr_php_wrap_user_function(NR_PSTR("GuzzleHttp\\Client::__construct"),
+                              nr_guzzle7_client_construct TSRMLS_CC);
+  }else {
```

Comment:
I think this code can be simplified further because `nr_guzzle6_client_construct` and `nr_guzzle7_client_construct` are actually `nr_guzzle_client_construct_helper`. Also I don't think the guzzle_version needs to be used in any of the New Relic instrumenting namespaces or class names. This would simplify the code and reduce memory allocations.

Reply:
Yes, that is a great point! I have simplified the code [40ab8cf](https://github.com/newrelic/newrelic-php-agent/pull/498/commits/40ab8cf08c7f02ac014fc301e188bc04c5c6ea97)